python/hello-world-gpu/client.py

```diff
@@ -0,0 +1,192 @@
+            session_id=session_id,
+        )
+
+        # Get the results ids
+        array1_id = results["array1"].result_id
+        array2_id = results["array2"].result_id
+        output_id = results["output"].result_id
+        payload_id = results["payload"].result_id
+
+        # Arrays
+        # Set the seed for reproducibility
+        np.random.seed(seed)
+        a_host = np.random.rand(size).astype(np.float32)
+        b_host = np.random.rand(size).astype(np.float32)
+
+        # Create the metadata (a result) and upload data at the same time
+        result_client.upload_result_data(
+            result_id=array1_id,
+            session_id=session_id,
+            result_data=NumpyArraySerializer(a_host).serialize(),
+        )
+        result_client.upload_result_data(
+            result_id=array2_id,
+            session_id=session_id,
+            result_data=NumpyArraySerializer(b_host).serialize(),
+        )
+        logger.info("data uploaded")
+
+        # Creating a NameIdDict instance
+        name_id_mapping = NameIdDict(
+            {
+                "array1": array1_id,
+                "array2": array2_id,
+                "output": output_id,
+                "threads_per_block": 1024,
+                "blocks_per_grid": (size + (1024 - 1)) // 1024,
+            }
+        )
+
+        # Serializing the instance
+        serialized_name_id_mapping = name_id_mapping.serialize()
+
+        result_client.upload_result_data(
+            result_id=payload_id,
+            session_id=session_id,
+            result_data=serialized_name_id_mapping,
+        )
+        logger.info("payload uploaded")
+
+        # Submit task with payload and result ids
+        task_client.submit_tasks(
+            session_id=session_id,
+            tasks=[
+                TaskDefinition(
+                    data_dependencies=[array1_id, array2_id],
+                    expected_output_ids=[output_id],
+                    payload_id=payload_id,
+                )
+            ],
+        )
+        logger.info("tasks submitted")
+
+        # Wait for task end and result availability
+        try:
+            events_client.wait_for_result_availability(
+                result_ids=[output_id], session_id=session_id
+            )
+        except Exception as e:
+            logger.error("An error occured", extra={"context": {"error": e}})
+        # Download result
+        try:
+            serialized_result = result_client.download_result_data(
+                output_id, session_id
+            )
+            final_result = NumpyArraySerializer.deserialize(serialized_result).array
+            logger.info(
+                "Result ready",
+                extra={"context": {"resultId": output_id, "data": final_result}},
+            )
+        except Exception as e:
+            logger.error("An error occured", extra={"context": {"error": e.details}})
+
+    logger.info("End Connection!")
+
+
+def main(args: List[str]) -> None:
```

Comment:
You don't need to pass the arguments as a list, as just calling parse_args with no args already take the arguments of the command line